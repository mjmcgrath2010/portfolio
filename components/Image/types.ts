import { DefaultTheme } from "styled-components";
import * as AspectRatio from "@radix-ui/react-aspect-ratio";

export interface ImageContainerProps
  extends AspectRatio.AspectRatioProps,
    AspectRatio.PrimitiveDivProps {
  theme: DefaultTheme;
}

interface ImageProps {
  src: string;
  ratio: number;
}

export default ImageProps;
