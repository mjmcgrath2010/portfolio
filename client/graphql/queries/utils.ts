/*
 * Example usage:
 *   export const getStaticProps = () => mapQueriesToProps([{ posts: getPosts }]);
 */
type QueryProps = {
  props: Object;
};

export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc: QueryProps, queryObj: Function) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();
          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            props: {
              error: "Oops, something went wrong",
            },
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
