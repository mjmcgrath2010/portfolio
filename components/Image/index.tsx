import React from "react";
import styled from "styled-components";
import * as AspectRatio from "@radix-ui/react-aspect-ratio";

import ImageProps, { ImageContainerProps } from "./types";

const Container = styled(AspectRatio.Root)<ImageContainerProps>`
  width: fit-content;
  border-radius: ${({ theme }) => theme.borderRadius};
  box-shadow: ${({ theme }) => theme.shadows.main};
  margin: 0;
  padding: 0;
`;

const StyledImage = styled.img`
  height: 100%;
  width: ;
  margin: 0;
  padding: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
`;

const Image = ({ src, ratio }: ImageProps) => (
  <Container ratio={ratio}>
    <StyledImage src={src} />
  </Container>
);

Image.defaultProps = {
  src: "https://via.placeholder.com/150",
  ratio: 1,
};

export default Image;
