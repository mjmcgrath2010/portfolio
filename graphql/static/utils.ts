export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc, queryObj) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();

          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            error: e.message,
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
