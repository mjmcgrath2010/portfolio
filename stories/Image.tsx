
/* eslint-disable import/no-anonymous-default-export */
import React from 'react';
import Image from '../components/Image';
import ImageProps from "../components/Image/types";

export default {
  title: 'Components/Image',
  component: Image,
  argTypes: {},
  parameters: {}
};

export const DefaultStory = ({ ...rest }: ImageProps) => <Image {...rest} />;

DefaultStory.args = {};

DefaultStory.storyName = 'Image - Default Story';