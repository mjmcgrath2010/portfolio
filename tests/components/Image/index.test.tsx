import Image from '@components/Image';

import { render, screen } from "@test/utils";

describe('<Image />', () => {
  it("matches the snapshot", () => {
    const view = render(<Image />);

    expect(view).toMatchSnapshot();
  });
});