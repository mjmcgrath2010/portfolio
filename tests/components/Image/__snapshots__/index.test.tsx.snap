// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 5px;
  box-shadow: hsl(0deg 0% 27% / 15%) 2px 3px 5px;
  margin: 0;
  padding: 0;
}

.c1 {
  height: 100%;
  margin: 0;
  padding: 0;
  border-radius: 5px;
}

<body>
    <div>
      <div
        data-radix-aspect-ratio-wrapper=""
        style="position: relative; width: 100%; padding-bottom: 100%;"
      >
        <div
          class="c0"
          style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
        >
          <img
            class="c1"
            src="https://via.placeholder.com/150"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 5px;
  box-shadow: hsl(0deg 0% 27% / 15%) 2px 3px 5px;
  margin: 0;
  padding: 0;
}

.c1 {
  height: 100%;
  margin: 0;
  padding: 0;
  border-radius: 5px;
}

<div>
    <div
      data-radix-aspect-ratio-wrapper=""
      style="position: relative; width: 100%; padding-bottom: 100%;"
    >
      <div
        class="c0"
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
      >
        <img
          class="c1"
          src="https://via.placeholder.com/150"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
